katcp/test/test_inspecting_client.py

```diff
@@ -155,6 +157,88 @@ def test_update_index(self):
         self.assertTrue(index['B'].get('_changed'))
 
 
+class TestInspectingClientInspect(tornado.testing.AsyncTestCase):
+    """Check that inspection populates the request/sensor index correctly"""
+
+    def setUp(self):
+        super(TestInspectingClientInspect, self).setUp()
+        self.server = DeviceTestServer('', 0)
+        start_thread_with_cleanup(self, self.server, start_timeout=1)
+        self.server_with_timeout_hints = DeviceTestServerWithTimeoutHints('', 0)
+        start_thread_with_cleanup(self, self.server_with_timeout_hints,
+                                  start_timeout=1)
+
+    def _get_server(self, hints):
+        """Return a running test server with or without request timeout hints
+
+        Parameters
+        ----------
+        hints : bool
+            Whether or not the server should have request timeout hints enabled
+
+        """
+        ServerClass = (DeviceTestServerWithTimeoutHints if hints
+                       else DeviceTestServer)
+        server = ServerClass('', 0)
+        start_thread_with_cleanup(self, server, start_timeout=1)
+        host, port = server.bind_address
+        return host, port, server
+
+    def _get_expected_request_index(self, server):
+        """Return expected request index data structure for given server instance
+
+        Parameters
+        ----------
+        server : :class:`DeviceServer` instance
+
+        """
+        hints = getattr(server, 'request_timeout_hints', {})
+        expected = {}
+        for req, handler in server._request_handlers.items():
+            expected[req] = {'name': req,
+                             'description': handler.__doc__,
+                             'timeout_hint': hints.get(req)}
+        return expected
+
+    @tornado.gen.coroutine
+    def _test_inspect_requests_no_timeout_hints(self, timeout_hints):
```

Comment:
This name isn't so clear to me, since it might use timeout hints, depending on the parameter.  Suggest shortening to `_test_inspect_requests`.